### DISCUSSION TITLE
VLAN selection on device ports limited to 'Site'

### DISCUSSION THREAD

**Discussion by PieterL75** [netbox-community/netbox#6707] [Q&A]
Hey,

Currently, there is a hard filter on what VLANs can be selected on device interfaces.
"The untagged VLAN (MYNAME (1069)) must belong to the same site as the interface's parent device, or it must be global"

For us, this is a very limiting filter.
We have vlans that strech across several countries and sites. (don't ask why, they are just there :-) )

As we have several separated layer2 domains, it is not possible to put the vlans in 'Global'

How can this filter/enforcement be removed ?

Pieter

**Comment by candlerb** [comment]
What version of Netbox?

Have you tried VLAN groups?  Recent versions of Netbox are able to assign VLAN groups to multiple sites.  From [2.11 release notes](https://github.com/netbox-community/netbox/releases/tag/v2.11.0):

> **Variable Scope Support for VLAN Groups (#5284)**
> 
> In previous releases, VLAN groups could be assigned only to a site. To afford more flexibility in conveying the true scope of an L2 domain, a VLAN group can now be assigned to a region, site group (new in v2.11), site, location, or rack. VLANs assigned to a group will be available only to devices and virtual machines which exist within its scope.
> 
> For example, a VLAN within a group assigned to a location will be available only to devices assigned to that location (or one of its child locations), or to a rack within that location.

**Reply by PieterL75** [reply]
I'm running v2.11.7.

The link you send is indeed what I am trying, but I must be misunderstanding the concept somehow....
So I have

Site groups
- MyCompany
  - Country1
  - Country2

Sites
- Site1 (Group Country1)
- Site2 (Group Country2)

My Vlan groups
- L2domain1 (site group MyCompany)

My devices
- device1 (site1)
- device2 (site2)

Reading that documentation, I would assume that when Device1 (Site1, part of Group Country1, part of group MyCompany) would be able to use vlans of the L2domain, that is part of Sitegroup MyCompany.
I must say, that I can SEE the vlans in the dropdown of the interface vlans, but when I select it and update, it throws the error

Pieter

**Reply by candlerb** [reply]
It sounds to me like this is not working as intended (i.e. it should either accept the vlan or not show it) - but I'm not really familiar with the details.

@jeremystretch: is the scenario as described supposed to work?  If so, it sounds like a good integration test case to add.

**Reply by PieterL75** [reply]
@candlerb @jeremystretch Wondering if you have some time to review this again ?
If my VLAN Group is of the scope 'Site-Group', and my device is member of a Site that belongs to that SiteGroup.
then should I be able to assign VLAN from that Site-Group to the device's interface ?
(see details above).

I still get an error "The untagged VLAN (SWMGT (3005)) must belong to the same site as the interface's parent device, or it must be global"
